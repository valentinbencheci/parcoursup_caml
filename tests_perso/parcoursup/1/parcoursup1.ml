(* ###################### - DESCRIPTION TEST - ###################### *)
(* Dans ce test, nous vérifions comment les candidats refusent 
consécutivement les places à l'université. Ainsi on peut voir si les 
algorithmes "refuser" et "nouveau_jour" fonctionnent correctement. *)
(* ################################################################## *)

open Test_parcoursup_utils

let session = nouvelle_session ()

let () = 
  ajoute_candidat session ~nom_candidat:"Adam";
  ajoute_candidat session ~nom_candidat:"Olivie";
  ajoute_candidat session ~nom_candidat:"Marie";
  ajoute_candidat session ~nom_candidat:"Pierre";
  ajoute_candidat session ~nom_candidat:"Alan";
  ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:2;
  ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Paris" ~capacite:1;
  ajoute_voeu session 
    ~rang_repondeur:(Some 1) 
    ~nom_candidat:"Adam" 
    ~nom_formation:"Université Côte d'Azur";
  ajoute_voeu session 
    ~rang_repondeur:(None) 
    ~nom_candidat:"Olivie" 
    ~nom_formation:"Université de Toulon";
  ajoute_voeu session 
    ~rang_repondeur:(Some 1) 
    ~nom_candidat:"Marie" 
    ~nom_formation:"Université de Paris";
  ajoute_voeu session 
    ~rang_repondeur:(Some 1) 
    ~nom_candidat:"Pierre" 
    ~nom_formation:"Université de Paris";
  ajoute_voeu session 
    ~rang_repondeur:(Some 1) 
    ~nom_candidat:"Alan" 
    ~nom_formation:"Université de Paris";
  ajoute_commission session 
    ~nom_formation:"Université Côte d'Azur" 
    ["Adam";"Olivie";"Marie";"Pierre";"Alan"]; 
  ajoute_commission session 
    ~nom_formation:"Université de Toulon" 
    ["Adam";"Olivie";"Marie";"Pierre";"Alan"]; 
  ajoute_commission session 
    ~nom_formation:"Université de Paris" 
    ["Adam";"Olivie";"Marie";"Pierre";"Alan"]; 
    
  reunit_commissions session;

  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Olivie";
  affiche_propositions_en_attente session "Olivie";
  affiche_voeux_en_attente session "Marie";
  affiche_propositions_en_attente session "Marie";
  affiche_voeux_en_attente session "Pierre";
  affiche_propositions_en_attente session "Pierre";
  affiche_voeux_en_attente session "Alan";
  affiche_propositions_en_attente session "Alan";

  renonce session 
    ~nom_candidat:"Marie" 
    ~nom_formation:"Université de Paris";
  nouveau_jour session;
  affiche_voeux_en_attente session "Marie";
  affiche_propositions_en_attente session "Marie";
  affiche_voeux_en_attente session "Pierre";
  affiche_propositions_en_attente session "Pierre";
  affiche_voeux_en_attente session "Alan";
  affiche_propositions_en_attente session "Alan";

  renonce session 
    ~nom_candidat:"Pierre" 
    ~nom_formation:"Université de Paris";
  nouveau_jour session;
  affiche_voeux_en_attente session "Marie";
  affiche_propositions_en_attente session "Marie";
  affiche_voeux_en_attente session "Pierre";
  affiche_propositions_en_attente session "Pierre";
  affiche_voeux_en_attente session "Alan";
  affiche_propositions_en_attente session "Alan";

  renonce session 
    ~nom_candidat:"Alan" 
    ~nom_formation:"Université de Paris";
  nouveau_jour session;
  affiche_voeux_en_attente session "Marie";
  affiche_propositions_en_attente session "Marie";
  affiche_voeux_en_attente session "Pierre";
  affiche_propositions_en_attente session "Pierre";
  affiche_voeux_en_attente session "Alan";
  affiche_propositions_en_attente session "Alan";

  renonce session 
    ~nom_candidat:"Alan" 
    ~nom_formation:"Université de Paris";
  nouveau_jour session;
  affiche_voeux_en_attente session "Marie";
  affiche_propositions_en_attente session "Marie";
  affiche_voeux_en_attente session "Pierre";
  affiche_propositions_en_attente session "Pierre";
  affiche_voeux_en_attente session "Alan";
  affiche_propositions_en_attente session "Alan";