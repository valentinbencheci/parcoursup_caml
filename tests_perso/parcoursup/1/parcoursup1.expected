nouvelle session
ajoute candidat Adam
ajoute candidat Olivie
ajoute candidat Marie
ajoute candidat Pierre
ajoute candidat Alan
ajoute formation Université Côte d'Azur (2 places)
ajoute formation Université de Toulon (1 places)
ajoute formation Université de Paris (1 places)
ajoute voeu de Adam pour Université Côte d'Azur (rang=1)
ajoute voeu de Olivie pour Université de Toulon
ajoute voeu de Marie pour Université de Paris (rang=1)
ajoute voeu de Pierre pour Université de Paris (rang=1)
ajoute voeu de Alan pour Université de Paris (rang=1)
commission de classement de Université Côte d'Azur : [Adam; Olivie; Marie; Pierre; Alan]
commission de classement de Université de Toulon : [Adam; Olivie; Marie; Pierre; Alan]
commission de classement de Université de Paris : [Adam; Olivie; Marie; Pierre; Alan]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur}
Voeux en attente de Olivie : {}
Propositions en attente de Olivie : {Université de Toulon}
Voeux en attente de Marie : {}
Propositions en attente de Marie : {Université de Paris}
Voeux en attente de Pierre : {Université de Paris(0)}
Propositions en attente de Pierre : {}
Voeux en attente de Alan : {Université de Paris(1)}
Propositions en attente de Alan : {}
Marie renonce à Université de Paris
nouveau jour
Voeux en attente de Marie : {}
Propositions en attente de Marie : {}
Voeux en attente de Pierre : {}
Propositions en attente de Pierre : {Université de Paris}
Voeux en attente de Alan : {Université de Paris(0)}
Propositions en attente de Alan : {}
Pierre renonce à Université de Paris
nouveau jour
Voeux en attente de Marie : {}
Propositions en attente de Marie : {}
Voeux en attente de Pierre : {}
Propositions en attente de Pierre : {}
Voeux en attente de Alan : {}
Propositions en attente de Alan : {Université de Paris}
Alan renonce à Université de Paris
nouveau jour
Voeux en attente de Marie : {}
Propositions en attente de Marie : {}
Voeux en attente de Pierre : {}
Propositions en attente de Pierre : {}
Voeux en attente de Alan : {}
Propositions en attente de Alan : {}
Alan renonce à Université de Paris
nouveau jour
Voeux en attente de Marie : {}
Propositions en attente de Marie : {}
Voeux en attente de Pierre : {}
Propositions en attente de Pierre : {}
Voeux en attente de Alan : {}
Propositions en attente de Alan : {}
